src/test/java/LoginTest.java

```diff
@@ -0,0 +1,86 @@
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pageObjects.AllSongsPage;
+import pageObjects.HomePage;
+import pageObjects.LoginPage;
+
+public class LoginTest extends BaseTest {
+
+    @Test
+    public void loginValidEmailPassword() {
+        provideEmail("pearl.estrada@testpro.io");
+        providePassword("April969!!");
+        clickLoginBtn();
+        WebElement avatarIcon = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[class='avatar']")));
+        Assert.assertTrue(avatarIcon.isDisplayed());
+    }
+
+    @Test
+    public void loginInvalidEmailValidPassword() {
+        navigateToPage();
+        provideEmail("email@test.com");
+        providePassword("April969!!");
+        clickLoginBtn();
+        Assert.assertEquals(driver.getCurrentUrl(), url);
+    }
+
+    @Test
+    public void loginValidEmailEmptyPassword() {
+        navigateToPage();
+        provideEmail("pearl.estrada@testpro.io");
+        providePassword(" ");
+        clickLoginBtn();
+        Assert.assertEquals(driver.getCurrentUrl(), url);
+    }
+
+    @Test(dataProvider = "NegativeLoginTestData", dataProviderClass = TestDataProvider.class)
+    public void negativeLoginTest(String email, String password) {
+        provideEmail(email);
+        providePassword(password);
+        clickLoginBtn();
+        String expectedUrl = "//https://qa.koel.app/";
+        Assert.assertEquals(driver.getCurrentUrl(), expectedUrl);
+        System.out.println("Just Testing Console");
+    }
+
+    @Test
+    public void loginValidEmailValidPasswordTest() {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+        loginPage.provideEmail("pearl.estrada@testpro.io");
+        loginPage.providePassword("April969!!");
+        loginPage.clickSubmit();
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+    }
+
+    @Test
+    public void loginValidEmailPasswordUsingPOM() {
+       LoginPage loginPage = new LoginPage(driver);
+       HomePage homePage = new HomePage(driver);
+       AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+       loginPage.provideEmail("pearl.estrada@testpro.io");
+       loginPage.providePassword("April969!!");
+       loginPage.clickSubmit();
+
+       Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+    }
+    @Test
+    public void loginValidEmailPasswordUsingPageFactory() {
+        LoginPage loginPage = new LoginPage(driver);
+        HomePage homePage = new HomePage(driver);
+        AllSongsPage allSongsPage = new AllSongsPage(driver);
+
+        loginPage.provideEmailToLogin("pearl.estrada@testpro.io");
+        loginPage.providePasswordToLogin("April969!!");
+        loginPage.clickLoginBtn();
+
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+    }
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it.

![image](https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/104761482/aa560195-a021-4196-a204-ca4676aa372b)

**HINT**

**How to analyze the error:**

Problems tool window (File Project Errors):
The Current File tab shows the list of problems in the file that is open in the editor. Select a problem and press Enter or double-click to navigate to it. You can also open a preview right in the tool window, and you can use the eye icon to filter the problems by type and to change the sorting.

You can select any problem in the window and press F4  or double-click it to jump to the corresponding line in the editor.

For example 
![image](https://github.com/Test-Pro-Bootcamp/QA-Automation-Framework-Template/assets/104761482/7caf54f1-0e32-4164-8d74-ca228119f97f)

The light bulb icon on the toolbar turns on if there are quick fixes for the selected problem. You can also press Alt+Enter on a problem to see the list of available quick fixes.

![image](https://github.com/Test-Pro-Bootcamp/QA-Automation-Framework-Template/assets/104761482/dd115f9c-3459-4476-9c70-2573ad220013)

![image](https://github.com/Test-Pro-Bootcamp/QA-Automation-Framework-Template/assets/104761482/450f92f3-1480-47e4-a951-25438bd30e49)

![image](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/e556f0ed-e934-4c91-971e-1835e3465c45)

https://github.com/Angelina0504/A53-QA-Automation-Framework/assets/104761482/56b1cb3a-8f19-4764-bc48-fe355d79d9d4